octree_creation_app/utils.py

```diff
@@ -220,12 +236,26 @@ def octree_2_treemesh(  # pylint: disable=too-many-locals
 
     :return: Resulting TreeMesh.
     """
-    if mesh.octree_cells is None:
+    if (
+        mesh.octree_cells is None
+        or mesh.u_count is None
+        or mesh.v_count is None
+        or mesh.w_count is None
+    ):
         return None
 
-    small_cell = [mesh.u_cell_size, mesh.v_cell_size, mesh.w_cell_size]
-    n_cell_dim = [mesh.u_count, mesh.v_count, mesh.w_count]
-    cell_sizes = [np.ones(nr) * sz for nr, sz in zip(n_cell_dim, small_cell)]
+    n_cell_dim, cell_sizes = [], []
+    for ax in "uvw":
+        if (
+            getattr(mesh, f"{ax}_cell_size") is None
+            or getattr(mesh, f"{ax}_count") is None
+        ):
+            raise ValueError(f"Cell size in {ax} direction is not defined.")
+
+        n_cell_dim.append(getattr(mesh, f"{ax}_count"))
+        cell_sizes.append(
+            np.ones(getattr(mesh, f"{ax}_count")) * getattr(mesh, f"{ax}_cell_size")
+        )
```

Comment:
**suggestion:** more readable if assigning after the loop, while loop is dedicated to safety checks

```suggestion
    n_cell_dim = [mesh.u_count, mesh.v_count, mesh.w_count]
    cell_sizes = [
        np.ones(mesh.u_count) * mesh.u_cell_size,
        np.ones(mesh.v_count) * mesh.v_cell_size,
        np.ones(mesh.w_count) * mesh.w_cell_size
    ]
```

Reply:
but that can be applied to develop. Still good to merge here

Reply:
Unfortunetly, the linters aren't that smart and don't take into account the getattr for the checks...